src/misc/nametag.ts

```diff
@@ -0,0 +1,33 @@
+import { PlayerJoinEvent } from 'org.bukkit.event.player';
+import { NameTagVisibility } from 'org.bukkit.scoreboard';
+
+const TEAMNAME = 'nametag';
+
+const sm = server.getScoreboardManager();
+
+const scoreboard = sm.getMainScoreboard();
+let team = scoreboard.getTeam(TEAMNAME);
+
+const initTeam = () => {
+  // Create new team, which hides nametag from the player
+  team = scoreboard.registerNewTeam(TEAMNAME);
+  team.setNameTagVisibility(NameTagVisibility.NEVER);
+};
+
+if (!team) {
+  initTeam();
+}
+
+registerEvent(PlayerJoinEvent, (event) => {
+  const player = event.getPlayer();
+
+  try {
+    team?.addPlayer(player);
+  } catch (error) {
```

Comment:
... or this might be completely unrelated error. Check that the team exists before you add players to it. If that fails, *additionally* catching the exception might make sense.